src/ngraph/op/util/scatter_base.cpp

```diff
@@ -0,0 +1,149 @@
+    const auto& axis_shape = get_input_partial_shape(AXIS);
+
+    NODE_VALIDATION_CHECK(this,
+                          axis_shape.compatible(PartialShape{}) ||
+                              axis_shape.compatible(PartialShape{1}),
+                          "Axis input shape is required to be scalar or 1D tensor. ",
+                          "Got: ",
+                          axis_shape);
+
+    // Updates rank must be at indices rank + data rank - 1
+    NODE_VALIDATION_CHECK(this,
+                          data_shape.rank().is_dynamic() || indices_shape.rank().is_dynamic() ||
+                              updates_shape.rank().is_dynamic() ||
+                              updates_shape.rank().get_length() ==
+                                  indices_shape.rank().get_length() +
+                                      data_shape.rank().get_length() - 1,
+                          "Updates rank is expected to be indices rank + data rank - 1.");
+
+    bool compatible = true;
+    int64_t axis;
+    bool is_axis_constant = input_value(AXIS).get_node_shared_ptr()->is_constant();
+
+    // Get axis value if possible.
+    if (is_axis_constant && data_shape.rank().is_static())
+    {
+        const auto axis_const_input =
+            as_type_ptr<op::v0::Constant>(input_value(AXIS).get_node_shared_ptr());
+        axis = axis_const_input->cast_vector<int64_t>().at(0);
+        axis = normalize_axis(this, axis, data_shape.rank().get_length());
+    }
+
+    if (is_axis_constant && data_shape.rank().is_static() && indices_shape.rank().is_static() &&
+        updates_shape.rank().is_static())
+    {
+        for (int64_t i = 0; i < indices_shape.rank().get_length(); ++i)
+        {
+            compatible = compatible && updates_shape[axis + i].compatible(indices_shape[i]);
+        }
+
+        int64_t indices_rank = indices_shape.rank().get_length();
+        // Check [d_0, d_1, ... d_(axis - 1)] updates dimensions
+        for (int64_t i = 0; i < axis; ++i)
+        {
+            compatible = compatible && updates_shape[i].compatible(data_shape[i]);
+        }
+        // Check [d_(axis + k + 1), ..., d_n] updates dimensions
+        for (int64_t i = axis + 1; i < data_shape.rank().get_length(); ++i)
+        {
+            compatible =
+                compatible && updates_shape[indices_rank - 1 + i].compatible(data_shape[i]);
+        }
+    }
+
+    NODE_VALIDATION_CHECK(this,
+                          compatible,
+                          "Updates shape must have appropriate dimensions equal to indices and "
+                          "data dimensions. Updates shape:",
+                          updates_shape.to_shape(),
+                          ", data shape: ",
+                          data_shape.to_shape(),
+                          ", indices_shape: ",
+                          indices_shape.to_shape(),
+                          ", axis: ",
+                          axis,
```

Comment:
- `to_shape` can thrown additonal exception during `NODE_VALIDATION_CHECK` assert
- Value of `axis` can be not defined

Reply:
I haven't updated this message after changing above `if` condition. This message is evaluated only when `compatible` is false, thus (earlier shapes where known) now only ranks are static for sure. However the axis value should also be known at this point.